src/lua/CommunityBalanceMod/FortressPvE/Crag.lua

```diff
@@ -0,0 +1,286 @@
+            techTree:SetTechNodeChanged(researchNode, string.format("researchProgress = %.2f", self.researchProgress)) 
+            
+        end
+
+    end
+
+
+    function Crag:OnResearchCancel(researchId)
+
+        if researchId == kTechId.UpgradeToFortressCrag then
+        
+            local team = self:GetTeam()
+            
+            if team then
+            
+                local techTree = team:GetTechTree()
+                local researchNode = techTree:GetTechNode(kTechId.Crag)
+                if researchNode then
+                    researchNode:ClearResearching()
+                    techTree:SetTechNodeChanged(researchNode, string.format("researchProgress = %.2f", 0))   
+                end
+            end  
+        end
+    end
+
+    -- Called when research or upgrade complete
+    function Crag:OnResearchComplete(researchId)
+
+        if researchId == kTechId.UpgradeToFortressCrag then
+        
+            self:UpgradeToTechId(kTechId.FortressCrag)
+            --UpdateHealthValues(newtechid)
+            --self:SetTechId(kTechId.FortressCrag)
+            
+            local techTree = self:GetTeam():GetTechTree()
+            local researchNode = techTree:GetTechNode(kTechId.Crag)
+            
+            if researchNode then     
+    
+                researchNode:SetResearchProgress(1)
+                techTree:SetTechNodeChanged(researchNode, string.format("researchProgress = %.2f", self.researchProgress))
+                researchNode:SetResearched(true)
+                techTree:QueueOnResearchComplete(kTechId.FortressCrag, self)
+
+                
+
+                
+            end
+            
+        end
+    end
+
+end
+
+
+if Client then
+    
+    function Crag:OnUpdateRender()
+
+           if not self.fortressCragMaterial and self:GetTechId() == kTechId.FortressCrag then
+ 
+                local model = self:GetRenderModel()
+
+                if model and model:GetReadyForOverrideMaterials() then
+                
+                    model:ClearOverrideMaterials()
+                    --local material = GetPrecachedCosmeticMaterial( "Crag", "Fortress" )
+                    local material = Crag.kfortressCragMaterial
+                    assert(material)
+                    model:SetOverrideMaterial( 0, material )
+
+                    self.fortressCragMaterial = true
+                end
+
+           end
+    end
+end
+
+
+function Crag:GetMatureMaxHealth()
+
+    if self:GetTechId() == kTechId.FortressCrag then
+        return kFortressMatureCragHealth
+    end
+
+    return kMatureCragHealth
+end
+
+
+function Crag:GetMatureMaxArmor()
+
+    if self:GetTechId() == kTechId.FortressCrag then
+        return kFortressMatureCragArmor
+    end
+
+    return kMatureCragArmor
+end    
+
+
+
+
+
+function Crag:OnAdjustModelCoords(modelCoords)
+    --gets called a ton each second
+
+    if not self.reduceSized and self:GetTechId() == kTechId.Crag then
```

Comment:
Is this supposed to be self.reduce**d**Sized? I don't see where self.reduceSized is set

Reply:
Corrected the typo. I have no idea why it even worked before without issues

Reply:
I know now. This was a 360 degrees miss